addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -112,67 +119,81 @@ public synchronized void playerStop() {
-
-        JsonObject params = new JsonObject();
-        params.addProperty("playerid", activePlayer);
-        params.addProperty("to", "next");
-        socket.callMethod("Player.GoTo", params);
+        internalGoTo("next");
 
         updatePlayerStatus();
     }
 
     public synchronized void playerPrevious() {
+        internalGoTo("previous");
+
+        updatePlayerStatus();
+    }
+
+    private void internalGoTo(@NonNull String to) {
         int activePlayer = getActivePlayer();
 
         JsonObject params = new JsonObject();
         params.addProperty("playerid", activePlayer);
-        params.addProperty("to", "previous");
+        params.addProperty("to", to);
         socket.callMethod("Player.GoTo", params);
-
-        updatePlayerStatus();
     }
 
-    public synchronized void playerRewind() {
+    private int calcNextSpeed(int modifier) {
         int activePlayer = getActivePlayer();
+        if (activePlayer >= 0) {
+            int position = SPEEDS.indexOf(getSpeed(activePlayer));
+            if (position == -1) {
+                return 0;
+            } else if (position == 0 || position == (SPEEDS.size() - 1)) {
+                return SPEEDS.get(position);
+            } else {
+                return (int) SPEEDS.get(position + modifier);
+            }
+        } else {
+            return 0;
+        }
+    }
 
-        JsonObject params = new JsonObject();
-        params.addProperty("playerid", activePlayer);
-        params.addProperty("speed", -2);
-        socket.callMethod("Player.SetSpeed", params);
+    public synchronized void playerRewind() {
+        internalSetSpeed(calcNextSpeed(-1));
 
         updatePlayerStatus();
     }
 
     public synchronized void playerFastForward() {
+        internalSetSpeed(calcNextSpeed(1));
+
+        updatePlayerStatus();
+    }
+
+    private void internalSetSpeed(int speed) {
         int activePlayer = getActivePlayer();
 
         JsonObject params = new JsonObject();
         params.addProperty("playerid", activePlayer);
-        params.addProperty("speed", 2);
+        params.addProperty("speed", speed);
         socket.callMethod("Player.SetSpeed", params);
-
-        updatePlayerStatus();
     }
 
     public synchronized void increaseVolume() {
         this.volume += VOLUMESTEP;
-        JsonObject params = new JsonObject();
-        params.addProperty("volume", volume);
-        socket.callMethod("Application.SetVolume", params);
+        internalSetVolume();
     }
 
     public synchronized void decreaseVolume() {
         this.volume -= VOLUMESTEP;
-        JsonObject params = new JsonObject();
-        params.addProperty("volume", volume);
-        socket.callMethod("Application.SetVolume", params);
+        internalSetVolume();
     }
 
     public synchronized void setVolume(int volume) {
         this.volume = volume;
+        internalSetVolume();
+    }
+
+    private void internalSetVolume() {
         JsonObject params = new JsonObject();
```

Comment:
can the actual setting of the volume be moved here? It feels more sensible if `internalSetVolume` would set the value